### ISSUE TITLE
[ BUG ] Mismatching execution languages in `Invoke-FalconDeploy`

### ISSUE THREAD

**Issue by MatthewCKelly** [CrowdStrike/psfalcon#327]
**Describe the bug**
When building string for the command line, it mixes languages that being set-location is a PowerShell verb and not a batch/command verb.
other issues

1. the single quote are not valid at the command prompt
2. the command "cmd" should be "cmd.exe"
 

**To Reproduce**
Create a batch file and add to zip
```batch
@echo off
@echo starting in :  "%~dp0"
PUSHD "%~dp0"
@Echo Argument 1= "%~1"
@Echo Argument 2= "%~2"
@Echo Current directory [%CD%]
@Echo Current Time %Time%

if Not Exist "C:\Program Files\CrowdStrike\CSSensorSettings.exe" (
Echo Missing Crowd Strike Falcon
EXIT 1

) else (
Echo CrowdStrike Falcon is installed.
)
PopD

```

Authenticate with service and then run 
```powershell
Invoke-FalconDeploy -Archive 'C:\PathToFile\SimpleBatchFile.zip' -Run 'SimpleBatchFile.cmd' -HostId 00000000000000000000`
```
**Expected behavior**
it changes to the temporary location and executes the file.

**Environment (please complete the following information):**
 - OS: Windows 10
 - PowerShell:  5.1
 - PSFalcon: 2.1.7

**Additional context**
from the source code "https://github.com/CrowdStrike/psfalcon/blob/master/public/psf-real-time-response.ps1"

```powershell
[string]$String = if ($Argument) {
    "Set-Location $TempDir;$($TempDir,$RunFile -join $Join)",
        $Argument -join ' '
} else {
    "Set-Location $TempDir;$($TempDir,$RunFile -join $Join)"
}


[string]$Executable = if ($RunFile -match '\.cmd$') {
    'cmd',"'/c $String'" -join ' '
} else {
    'powershell.exe',"'-c &{$String}'" -join ' '
}
```

So when the the variable String is created, is is assumed that it should be using 'Set-Location' however that is not a known verb at the command line..

test case code  
```powershell
$Join = '\';
$TempDir = '\windows\temp\FalconDeploy_20230627T194451459';

$Argument = '';
$RunFile  = 'SimpleBatchFile.cmd';

[string]$String = if ($Argument) {
    "Set-Location $TempDir;$($TempDir,$RunFile -join $Join)",
        $Argument -join ' '
} else {
    "Set-Location $TempDir;$($TempDir,$RunFile -join $Join)"
}

$String
# Set-Location \windows\temp\FalconDeploy_20230627T194451459;\windows\temp\FalconDeploy_20230627T194451459\SimpleBatchFile.cmd


[string]$Executable = if ($RunFile -match '\.cmd$') {
    'cmd',"'/c $String'" -join ' '
} else {
    'powershell.exe',"'-c &{$String}'" -join ' '
}

$Executable
# cmd '/c Set-Location \windows\temp\FalconDeploy_20230627T194451459;\windows\temp\FalconDeploy_20230627T194451459\SimpleBatchFile.cmd'
```

Updated code - how I would do it..
```powershell
[string]$String = if ($Argument) {
    "Set-Location $TempDir;$(Join-Path -Path $TempDir -ChildPath $RunFile)",
        $Argument -join ' '
} else {
    "Set-Location $TempDir;$(Join-Path -Path $TempDir -ChildPath $RunFile)"
}

Write-verbose "$String"

[string]$Executable = if ( ($RunFile -match '\.cmd$') -or ($RunFile -match '\.bat$' )) {
    "cmd.exe /c cd /d `"$TempDir`" && `"$RunFile`" $Argument"
} else {
    'powershell.exe',"'-c &{$String}'" -join ' '
}

'-Raw=```Start-Process',$Executable,
"-RedirectStandardOutput '$(Join-Path -Path $TempDir -ChildPath stdout.log)'",
"-RedirectStandardError '$(Join-Path -Path $TempDir -ChildPath stderr.log)'",
('-PassThru | ForEach-Object { "The process was successfully sta' +
'rted"'),'}```' -join ' '

Write-verbose "$Executable"
```
this code would produce the following..
```powershell
# $String = 
Set-Location \windows\temp\FalconDeploy_20230627T194451459;\windows\temp\FalconDeploy_20230627T194451459\SimpleBatchFile.cmd Hello world foo Bar

# $Executable =
-Raw=```Start-Process cmd.exe /c cd /d "\windows\temp\FalconDeploy_20230627T194451459" && "SimpleBatchFile.cmd" Hello world foo Bar -RedirectStandardOutput '\windows\temp\FalconDeploy_20230627T194451459\stdout.log' -RedirectStandardError '\wi
ndows\temp\FalconDeploy_20230627T194451459\stderr.log' -PassThru | ForEach-Object { "The process was successfully started" }```
```

**Transcript content**

>VERBOSE: 19:45:06 [ApiClient.Invoke] {"optional_hosts":["6DDC85A541B049B99C1CC4661E9AC732"],"command_string":"runscript -Raw=```Start-Process cmd \u0027/c Set-Location \\Windows\\Temp\\FalconDeploy_20230627T1944514595;\\Windows\\Temp\\FalconDeploy_20230627T1944514595\\\u0026SimpleBatchFile.cmd\u0027 -RedirectStandardOutput \u0027\\Windows\\Temp\\FalconDeploy_20230627T1944514595\\stdout.log\u0027 -RedirectStandardError \u0027\\Windows\\Temp\\FalconDeploy_20230627T1944514595\\stderr.log\u0027 -PassThru | ForEach-Object { \"The process was successfully started\" }```","batch_id":"24717A60-2C45-4C8D-95B3-C02C1FC15DC7","base_command":"runscript"}

**Comment by bk-cs** [comment]
Thank you! I'm on vacation but I'll take a look at this as soon as I'm back. Seems like a thorough solution to something that I didn't catch.

**Comment by bk-cs** [comment]
@MatthewCKelly

Thank you for the advice! I propose the following change to `psf-real-time-response.ps1`. Would you like to commit this change to the 2.2.6 branch for recognition?

Current code:
```powershell
[string]$String = if ($Argument) {
  "Set-Location $TempDir;$($TempDir,$RunFile -join $Join)",
  $Argument -join ' '
} else {
  "Set-Location $TempDir;$($TempDir,$RunFile -join $Join)"
}
```
New code:
```powershell
[string]$String = if ($RunFile -match '\.cmd$') {
  'cd /d "{0}" && "{1}"' -f $TempDir,$RunFile
} else {
  "Set-Location $TempDir;$($TempDir,$RunFile -join $Join)"
}
if ($Argument) { $String = $String,$Argument -join ' ' }
```

**Comment by MatthewCKelly** [comment]
I've got some other updates as well, away from office so will submit once I'm online again...

**Comment by bk-cs** [comment]
> I've got some other updates as well, away from office so will submit once I'm online again...

Would you still like to pull and commit this change? If you're unable to do so, I can put a shout out in the release notes instead. :)

**Comment by bk-cs** [comment]
Reopening issue for awareness until 2.2.6 is deployed.

I merged your code into `master` and then into `2.2.6`. I made a few minor changes to make the spacing match the `2.2.6` release (2 character indent instead of 4 to reduce file size), took out the additional `powershell.exe` parameters (because they don't have an impact with how PowerShell is executed via RTR), and removed the `Join-Path` entries because I wanted the joining to match how it happens in the rest of the script... even though it's against PowerShell style. :)

**Comment by MatthewCKelly** [comment]
Greetings,  the Powershell Parameters to have an impact.. Hence my change.. I can log another issue for that ?

**Comment by bk-cs** [comment]
It was working as expected in my testing--are you finding something different?

**Comment by bk-cs** [comment]
I believe this is resolved in 2.2.6 release which has been published to GitHub and will be published on the PowerShell Gallery soon.
